calyx-py/test/correctness/queues/strict_queues/strict.py

```diff
@@ -0,0 +1,235 @@
+# pylint: disable=import-error
+import os
+import sys
+import inspect
+
+currentdir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+parentdir = os.path.dirname(currentdir)
+sys.path.insert(0, parentdir)
+
+import fifo
+import calyx.builder as cb
+import calyx.queue_call as qc
+
+# This determines the maximum possible length of the queue:
+# The max length of the queue will be 2^QUEUE_LEN_FACTOR.
+QUEUE_LEN_FACTOR = 4
+
+
+def invoke_subqueue(queue_cell, cmd, value, ans, err) -> cb.invoke:
+    """Invokes the cell {queue_cell} with:
+    {cmd} passed by value
+    {value} passed by value
+    {ans} passed by reference
+    {err} passed by reference
+    """
+    return cb.invoke(
+        queue_cell,
+        in_cmd=cmd,
+        in_value=value,
+        ref_ans=ans,
+        ref_err=err,
+    )
+
+
+def insert_strict_pifo(
```

Comment:
Super cool! I have a high-level suggestion before I dive in with a full review: would you like to try and refactor `strict` and `roundrobin`'s eDSL files so that they both draw from some common file? They clearly share a lot of code, and I bet you could cleverly write some common code that is parameterized based on what algorithm is being run. I think just the logic for changing `hot` will change, right?

The benefit of doing this is maintainability. I feel like we're learning more about how to use the eDSL all the time, and it would be nice to have similar code executed similarly. To be clear, the way you have done it is exactly the correct first step! Now that we know that this works, I feel like you will have an easier time implementing this general parameterized solution that I am thinking of!